core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineSqlServer.java

```diff
@@ -223,14 +223,16 @@ public DataType getDataTypeForScalarDb(
         return DataType.TEXT;
       case BINARY:
       case VARBINARY:
-        if (typeName.equalsIgnoreCase("timestamp") || typeName.equalsIgnoreCase("hierarchyid")) {
+        if (!typeName.equalsIgnoreCase("binary") && !typeName.equalsIgnoreCase("varbinary")) {
```

Comment:
This is the main part of this fix. A black-list-based exclusion sometimes makes us miss the unsupported data types. So, we only accept the expected ones.